backend/api/oauthApi.ts

```diff
@@ -1,38 +1,49 @@
+import axios, { AxiosError, AxiosInstance } from 'axios'
+import querystring from 'querystring'
+import logger from '../../logger'
+import errorStatusCode from '../error-status-code'
+
 /** @type {any} */
-const axios = require('axios')
-const querystring = require('querystring')
-const logger = require('../log')
-const errorStatusCode = require('../error-status-code')
+export const AuthClientErrorName = 'AuthClientError'
+export const AuthClientError = (message: string) => ({ name: AuthClientErrorName, message, stack: new Error().stack })
 
-const AuthClientErrorName = 'AuthClientError'
-const AuthClientError = (message) => ({ name: AuthClientErrorName, message, stack: new Error().stack })
+export const apiClientCredentials = (clientId: string, clientSecret: string) =>
+  Buffer.from(`${clientId}:${clientSecret}`).toString('base64')
 
-const apiClientCredentials = (clientId, clientSecret) => Buffer.from(`${clientId}:${clientSecret}`).toString('base64')
+export interface OAuthApi {
+  refresh: (refreshToken: string) => Promise<{ access_token: string; refresh_token: string }>
+  oauthAxios: AxiosInstance
+}
+
+export interface OAuthApiFactoryParams {
+  apiClientId: string
+  apiClientSecret: string
+  url: string
+}
 
 /**
  * Return an oauthApi built using the supplied configuration.
- * @param client
  * @param {object} params
- * @param {string} params.clientId
- * @param {string} params.clientSecret
- * @param {string} params.url
  * @returns a configured oauthApi instance
  */
-const oauthApiFactory = (client, { clientId, clientSecret, url }) => {
-  const oauthAxios = axios.create({
-    baseURL: `${url}/oauth/token`,
+export const oauthApiFactory = (params: OAuthApiFactoryParams): OAuthApi => {
```

Comment:
this is the change in signature referred to above

as said I have removed `client` from parameters which was unused and created a dependency on oauthEnabledClient